### PR TITLE
Closes #80 | Implement MSVD-Indonesian Dataloader

### LINKED ISSUE #80: Create dataset loader for MSVD-Indonesian
Dataloader name: `id_msvd/id_msvd.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?id_msvd

|     Dataset| id_msvd |
|-------------|---|
| Description | MSVD-Indonesian is derived from the MSVD (Microsoft Video Description) dataset, which is obtained with the help of a machine translation service (Google Translate API). This dataset can be used for multimodal video-text tasks, including text-to-video retrieval, video-to-text retrieval, and video captioning. Same as the original English dataset, the MSVD-Indonesian dataset contains about 80k video-text pairs. |
| Subsets | - |
| Languages | ind |
| Tasks | Text Retrieval, Image-to-Text Generation |
| License | MIT (mit) |
| Homepage | https://github.com/willyfh/msvd-indonesian |
| HF URL | - |
| Paper URL | https://arxiv.org/abs/2306.11341 |

### PR DESCRIPTION
**Pull request by akhdanfadh**
Closes #80 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/id_msvd/id_msvd.py
Comment (sabilmakbar) [reply seacrowd/sea_datasets/id_msvd/id_msvd.py:127]: Furthermore, for the context of a Retrieval System, the data is constructed by creating a pair of query-document with a label (positive or negative). In this context a pair can be constructed by a video and a text; imagine searching a video on YT based on its video captioning rather than its title.

Therefore, the schema for VT2R should be extended to accommodate the label for a given pair (although it's all positive pairs in this dataset).

cc @holylovenia @SamuelCahyawijaya

**Review by sabilmakbar — COMMENTED** [review]
a nit & additional comment on the implementation

File: seacrowd/sea_datasets/id_msvd/id_msvd.py
Comment (akhdanfadh) [reply seacrowd/sea_datasets/id_msvd/id_msvd.py:127]: I see, then there will be several pairs for a video then.
I mistakenly assumed that `texts` field in video schema is a list, thanks for pointing that out.

I'm not sure about the general application of a Retrieval System in NLP. In video-text retrieval, the goal is mainly to find the most relevant video related to the semantics in a given sentence, and vice versa - with no text label in general. If we want to accommodate something (including labeling such as domain or video categories, etc), we can add `context` field as a child in `metadata`.

Comment (holylovenia) [reply seacrowd/sea_datasets/id_msvd/id_msvd.py:127]: Hi @akhdanfadh and @sabilmakbar, sorry I missed this discussion. We can use the `video` schema for this. The `texts` field does not necessarily refer to transcribed utterances, it could also refer to relevant queries as you said.

Comment (akhdanfadh) [reply seacrowd/sea_datasets/id_msvd/id_msvd.py]: 8d8c04a : Changed the loading scheme based on the above, but I left the source as is. Also, adding a minor change to the key name in the `video` schema for clarity.

Comment (sabilmakbar) [reply seacrowd/sea_datasets/id_msvd/id_msvd.py]: cool, thanks and welcome back, @akhdanfadh! at a glance it looks okay, I'll test your newly implemented dataloader tonight. One thing, mind separating the video schema minor changes into a different PR? thanks in adv!

File: seacrowd/sea_datasets/id_msvd/id_msvd.py
Comment (sabilmakbar) [inline seacrowd/sea_datasets/id_msvd/id_msvd.py:75]: in source, the data is a pair of video_path and single text
```suggestion
                    "text": datasets.Value("string"),
```

File: seacrowd/sea_datasets/id_msvd/id_msvd.py
Comment (sabilmakbar) [inline seacrowd/sea_datasets/id_msvd/id_msvd.py:122]: hence, in SEACrowd Source Implementation, we shouldn't group by its video path, because it will alter it from the source data
```suggestion
            for i, row in df.iterrows():
                yield i, {
                    "video_path": str(row["video_path"]),
                    "text": row["text"],
```

**Review by sabilmakbar — CHANGES_REQUESTED** [review]
A review on implementation on source, pls refer to this initial [data source](https://raw.githubusercontent.com/willyfh/msvd-indonesian/main/data/MSVD-indonesian.txt) to match with SEACrowd Source Implementation.

One more thing, mind separating the video schema minor changes into a different PR? thanks in adv!

**Comment by akhdanfadh** [comment]
Hey, thanks for the review. Here's what I'll do, expect to be done this Friday night.
- Not sure why in the first place I aggregate the video path 😅. Sure I will change it according to the source.
- I will leave the `texts` key as it is and after this PR merged, I'll create a new PR for changing that key and in the two dataloaders accordingly.

**Review by sabilmakbar — APPROVED** [review]
Tested and all good, LGTM! Thanks @akhdanfadh, let's wait for @yongzx review.

**Comment by holylovenia** [comment]
Replacing @yongzx with @ljvmiranda921 due to the lack of response.

**Review by ljvmiranda921 — COMMENTED** [review]
One small nit for `make check_file=...` and then we're good to merge! Lmk if you have trouble running it.

**Comment by akhdanfadh** [comment]
@ljvmiranda921 Done! I personally don't like the changes, but I'm all good with it. Thanks!

**Review by ljvmiranda921 — APPROVED** [review]

**Comment by ljvmiranda921** [comment]
Well, gotta make sure the formatting's consistent. Thanks for updating!

**Pull request merged**
